x-pack/test/usage_collection/plugins/application_usage_test/public/plugin.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Plugin, CoreSetup, CoreStart } from 'kibana/public';
+import { first } from 'rxjs/operators';
+import expect from '@kbn/expect';
+import { applicationUsageSchema } from '../../../../../../src/plugins/kibana_usage_collection/server/collectors/application_usage/schema';
+
+export class ApplicationUsageTest implements Plugin {
+  public setup(core: CoreSetup) {}
+
+  public async start(core: CoreStart) {
+    if (core.http.anonymousPaths.isAnonymous(window.location.pathname)) {
+      return;
+    }
+    const applications = await core.application.applications$.pipe(first()).toPromise();
+    const appIds = [...applications.keys()].sort();
+
+    try {
+      // When the lists don't match, the entire page load will fail.
+      expect(Object.keys(applicationUsageSchema).sort()).to.eql(appIds);
+    } catch (err) {
+      err.message = `Application Usage's schema is not up-to-date with the actual registered apps. Please update it at src/plugins/kibana_usage_collection/server/collectors/application_usage/schema.ts.\n${err.message}`;
+      throw err;
+    }
```

Comment:
Instead of doing throwing an error on startup, it may be nicer to have this plugin set a property on `window.__usageCollectionTest__` or similar and have the test read that property. That way, other tests that we add to this suite can still run even if this test is currently failing.

Reply:
That makes perfect sense! In fact, I had a thought about it, but couldn't find a way to read the `window` object in the tests and thought it was not possible. Now that you mentioned, I ran a second search and found the right way to do it. Thanks! I'll push the changes in a bit.

Re the CI, I've checked Jenkins' logs to make sure the test runs. And I tried locally the fail scenario in the test. I'll push a failed scenario anyway to ensure we are setting things right by seeing the CI fail 👍

Reply:
The CI failed as expected: https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99129/testReport/ 🎉 

Running a last build with the correct schema.